### PR TITLE
editoast: replace NoTrainIdError by an empty list

### LINKED ISSUE #5448: stdcm: scheduling a train in an empty timetable doesn't show any result
### What happened?

I tried an stdcm request in an empty timetable. The request succeeded, but the follow-up requests failed: 

`https://[osrd]/editoast/train_schedule/results/?path_id=84&timetable_id=6`
```json
{
    "status": 400,
    "type": "editoast:train_schedule:NoTrainIds",
    "context": {},
    "message": "No train ids given"
}
```

It also seems to happen when there are trains in the timetable but no train appears on the given path. 

### What did you expect to happen?

A space-time chart with the new train, no error

### How can we reproduce it (as minimally and precisely as possible)?

1. Create a new timetable
2. Send an STDCM request

### What operating system, browser and environment are you using?

- **Browser**: Firefox 
- **Env**: dev

### OSRD version (top right corner `Account` button > `Informations`)

7396ff92

### PR DESCRIPTION
**Pull request by younesschrifi**
close #5448

### CODE REVIEW

**Review by leovalais — APPROVED** [review]
LGTM. Is there no front-end modifications to make? Maybe a unit test would be good

**Review by hamz2a — COMMENTED** [review]
LGTM

File: editoast/src/views/train_schedule/mod.rs [hunk 1]
```diff
@@ -731,3 +728,33 @@ pub fn process_simulation_response(
...
+        let url = format!(
+            "/train_schedule/results/?path_id={}&timetable_id={}",
+            pathfinding.id(),
+            scenario_fixture_set.timetable.id()
+        );
+        let req = TestRequest::get().uri(url.as_str()).to_request();
+        let response = call_service(&app, req).await;
+        assert_eq!(response.status(), StatusCode::OK);
+        let empty_vec: Vec<SimulationReport> = read_body_json(response).await;
+        assert_eq!(empty_vec, vec![])
```
Comment (leovalais) [inline editoast/src/views/train_schedule/mod.rs]:
```suggestion
        assert!(empty_vec.is_empty());
```

**Review by leovalais — APPROVED** [review]
LGTM

**Review by hamz2a — APPROVED** [review]

**Pull request merged**